server/webhook.go

```diff
@@ -38,145 +45,327 @@ type Webhook struct {
 			Status struct {
 				Id string
 			}
+			Labels []string
 		}
 	}
-	User struct {
-		Name        string
-		AvatarUrls  map[string]string
-		DisplayName string
-	}
+	User    WebhookUser
 	Comment struct {
-		Body string
+		Body         string
+		UpdateAuthor WebhookUser
 	}
 	ChangeLog struct {
 		Items []struct {
+			From       string
 			FromString string
+			To         string
 			ToString   string
 			Field      string
 		}
 	}
+	IssueEventTypeName string `json:"issue_event_type_name"`
+	RawJSON            string
+	showDescription    bool
 }
 
-// SlackAttachment returns the text to be placed in the resulting post or an empty string if nothing should be
-// posted.
-func (w *Webhook) SlackAttachment() (*model.SlackAttachment, error) {
-	switch w.WebhookEvent {
-	case "jira:issue_created":
-	case "jira:issue_updated":
-		isResolutionChange := false
-		for _, change := range w.ChangeLog.Items {
-			if change.Field == "resolution" {
-				isResolutionChange = (change.FromString == "") != (change.ToString == "")
-				break
-			}
+const (
+	md_RootStyle   = "## "
+	md_UpdateStyle = "###### "
+)
+
+var formatters = map[string]func(w *Webhook) string{
+	"jira:issue_created": mdIssueCreated,
+	"jira:issue_updated": mdIssueUpdated,
+	"comment_created":    mdComment,
+	"comment_updated":    mdComment,
+	"comment_deleted":    mdComment,
```

Comment:
Interesting way to do this :+1:

Reply:
I'll change it back to a switch actually, now that you pointed it out. I was anticipating more different event types when I set it up, but ended up with only a handful. a 'switch' would be consistent with other places.